This patch is based on the upstream commit described below, adjusted for use
in the Debian package by Peter Michael Green.

commit eb7fbc57d0acb52cf60bd86f01b76d849cd0d23a
Author: gwenn <gtreguier@gmail.com>
Date:   Sat Mar 25 10:19:56 2023 +0100

    Use ParseCallbacks::item_name to fix sqlite3_index_info_* types

Index: rusqlite/src/vtab/mod.rs
===================================================================
--- rusqlite.orig/src/vtab/mod.rs
+++ rusqlite/src/vtab/mod.rs
@@ -629,7 +629,7 @@ impl IndexConstraintUsage<'_> {
 
 /// `feature = "vtab"`
 pub struct OrderByIter<'a> {
-    iter: slice::Iter<'a, ffi::sqlite3_index_info_sqlite3_index_orderby>,
+    iter: slice::Iter<'a, ffi::sqlite3_index_orderby>,
 }
 
 impl<'a> Iterator for OrderByIter<'a> {
@@ -647,7 +647,7 @@ impl<'a> Iterator for OrderByIter<'a> {
 }
 
 /// A column of the ORDER BY clause.
-pub struct OrderBy<'a>(&'a ffi::sqlite3_index_info_sqlite3_index_orderby);
+pub struct OrderBy<'a>(&'a ffi::sqlite3_index_orderby);
 
 impl OrderBy<'_> {
     /// Column number
Index: rusqlite/Cargo.toml
===================================================================
--- rusqlite.orig/Cargo.toml
+++ rusqlite/Cargo.toml
@@ -96,7 +96,7 @@ version = "1.4"
 optional = true
 
 [dependencies.libsqlite3-sys]
-version = "0.25.0"
+version = "0.26.0"
 
 [dependencies.serde_json]
 version = "1.0"
@@ -146,8 +146,8 @@ features = ["v4"]
 
 [features]
 array = ["vtab"]
-backup = ["libsqlite3-sys/min_sqlite_version_3_6_23"]
-blob = ["libsqlite3-sys/min_sqlite_version_3_7_7"]
+backup = ["libsqlite3-sys/min_sqlite_version_3_14_0"]
+blob = ["libsqlite3-sys/min_sqlite_version_3_14_0"]
 buildtime_bindgen = ["libsqlite3-sys/buildtime_bindgen"]
 collation = []
 column_decltype = []
@@ -156,7 +156,7 @@ csvtab = [
     "vtab",
 ]
 extra_check = []
-functions = ["libsqlite3-sys/min_sqlite_version_3_7_7"]
+functions = ["libsqlite3-sys/min_sqlite_version_3_14_0"]
 hooks = []
 i128_blob = []
 limits = []
@@ -187,16 +187,16 @@ modern-full = [
     "window",
 ]
 modern_sqlite = ["libsqlite3-sys/buildtime_bindgen"]
-release_memory = ["libsqlite3-sys/min_sqlite_version_3_7_16"]
+release_memory = ["libsqlite3-sys/min_sqlite_version_3_14_0"]
 series = ["vtab"]
 session = [
     "libsqlite3-sys/session",
     "hooks",
 ]
 sqlcipher = ["libsqlite3-sys/sqlcipher"]
-trace = ["libsqlite3-sys/min_sqlite_version_3_6_23"]
+trace = ["libsqlite3-sys/min_sqlite_version_3_14_0"]
 unlock_notify = ["libsqlite3-sys/unlock_notify"]
-vtab = ["libsqlite3-sys/min_sqlite_version_3_7_7"]
+vtab = ["libsqlite3-sys/min_sqlite_version_3_14_0"]
 wasm32-wasi-vfs = ["libsqlite3-sys/wasm32-wasi-vfs"]
 window = ["functions"]
 with-asan = ["libsqlite3-sys/with-asan"]
